source/uwp/Renderer/lib/InputValue.cpp

```diff
@@ -253,40 +253,34 @@ HRESULT NumberInputValue::RuntimeClassInitialize(_In_ IAdaptiveNumberInput* adap
 HRESULT NumberInputValue::IsValueValid(_Out_ boolean* isInputValid)
```

Comment:
Is this something that can eventually land as part of shared code in the shared model at all?

Reply:
We could theoretically put a helper method for this and/or other validation methods in the shared model. I'm not sure how much we'd get out of that, because some portion of what's going on here is getting things out of xaml controls and whatnot, and what we have in hand is a UWP number input object not a shared model one. We could convert back, or just get the min and max values to pass to a helper, but the actual shared code is probably just a few not very interesting lines. (Although, admittedly, I'm fixing a consistency bug in those few lines so i guess they're at least a little interesting :)). 

Probably not worth doing now, but it's something we can keep in mind if we start seeing more consistency issues in this logic or if our local validation becomes more complex.

---
In reply to: [457623190](https://github.com/microsoft/AdaptiveCards/pull/4432#discussion_r457623190) [](ancestors = 457623190)